sycl/doc/extensions/experimental/sycl_ext_intel_data_flow_pipes_properties.asciidoc

```diff
@@ -0,0 +1,274 @@
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+== Version
+
+Built On: {docdate} +
+Revision: A
+
+== Contact
+
+Robert Ho, Intel (robert 'dot' ho 'at' intel 'dot' com)
+
+== Contributors
+
+Bo Lei, Intel +
+Marco Jacques, Intel +
+Joe Garvey, Intel +
+Aditi Kumaraswamy, Intel +
+Robert Ho, Intel +
+Sherry Yuan, Intel +
+Peter Colberg, Intel +
+Zibai Wang, Intel
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 6 and
+the following extensions:
+
+- link:../supported/sycl_ext_intel_dataflow_pipes.asciidoc[SYCL_INTEL_data_flow_pipes]
+- link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties]
+
+== Overview
+
+This extension introduces properties that establish differences in the
+implementation of `sycl::ext::intel::experimental::pipe`. These properties are FPGA specific. An example
+of the syntax can be seen below. 
+
+[source,c++]
+----
+using pipe = pipe<class some_pipe, int, min_capacity, decltype(properties{uses_valid<true>})>;
+----
+
+== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros". Therefore, an implementation
+supporting this extension must predefine the macro
+`SYCL_EXT_INTEL_FPGA_PIPE_PROPERTIES` to one of the values defined in the table
+below. Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's APIs the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version.  Base features are supported.
+|===
+
+=== Pipe properties
+
+Below is a list of compile-time-constant properties which `pipe` supports.
+
+```c++
+namespace sycl::ext::intel::experimental {
+
+struct ready_latency_key {
+  template <int Latency>
+  using value_t = oneapi::experimental::property_value<
+      ready_latency_key, std::integral_constant<int, Latency>>;
+};
+
+struct bits_per_symbol_key {
+  template <int Bits>
+  using value_t =
+      oneapi::experimental::property_value<bits_per_symbol_key,
+                                           std::integral_constant<int, Bits>>;
+};
+
+struct uses_valid_key {
+  template <bool Valid>
+  using value_t =
+      oneapi::experimental::property_value<uses_valid_key,
+                                           std::bool_constant<Valid>>;
+};  
+    
+struct first_symbol_in_high_order_bits_key {
+  template <bool HighOrder>
+  using value_t =
+      oneapi::experimental::property_value<first_symbol_in_high_order_bits_key,
+                                           std::bool_constant<HighOrder>>;
+};
+
+enum class protocol_name : std::uint16_t {
+  AVALON_STREAMING = 0,
+  AVALON_STREAMING_USES_READY = 1,
+  AVALON_MM = 2,
+  AVALON_MM_USES_READY = 3
```

Comment:
Upper-case enum values is a little unorthodox for a SYCL enum.

Reply:
I see that we already have them in the headers with the upper-case names. @gmlueck - Do you have a preference here? Should we keep the uppercase names or rename it here and in the headers?

Reply:
I agree that lower case is the norm for SYCL.  If we are just developing this extension now, I think it would be better to change to lower case.

If the extension has already been released and customer code is using the upper case spelling, we could just keep that.  I think it's somewhat up to the FPGA team.  If they want this extension to match stylistically, we could deprecate the upper case spellings and add the lower case spellings now.

Reply:
There's no existing customer code as this is the first release, so I think we're fine with changing the header to use lower case now.

Reply:
> Does the header change need to be part of a different PR?

I'm fine with either having it as part of this or as a follow-up patch. 😄

Reply:
Ok, I've added the protocol switch to lower-case in the header to this PR. The header also contained a couple of extraneous properties that came out of an earlier draft of the spec, so I removed those as well.